src/util/datablock/datablock_iterator.c

```diff
@@ -6,9 +6,16 @@
 */
 
 #include "datablock_iterator.h"
+#include "datablock.h"
 #include "assert.h"
 #include <stdio.h>
 
+int static inline _IsItemDeleted(int itemSize, char* item) {
```

Comment:
This function can be simplified a fair bit, I think (and we can definitely return once we encounter a byte that's not all 1s). One option:
```
int static inline _IsItemDeleted(int itemSize, unsigned char* item) {
  for(int i = 0; i < itemSize; i++) {
    if (item[i] != DELETED_MARKER) return 0;
  }
 return 1;
```
Note that this doesn't work without the change to `unsigned char *item`, which is a change I'd recommend making in general for these deleted item functions. The comparison behavior gets weird (and possibly compiler-specific) otherwise:
```
(gdb) p item[i]
$2 = -1 '\377'
(gdb) p (unsigned char)item[i]
$3 = 255 '\377'
(gdb) p item[i] == 0xff
$4 = 0
(gdb) p (unsigned char)item[i] == 0xff
$7 = 1
```

Reply:
👍